app/views/unattended/provisioning_templates/PXEGrub/pxegrub_default_local_boot.erb

```diff
@@ -4,8 +4,12 @@ name: PXEGrub default local boot
 model: ProvisioningTemplate
 %>
 <%# Used to boot provisioned hosts, do not associate or change the name. %>
-
-default=0
+<%
+  # Grub1 only supports numeric default statement, this array allows conversion to number
+  items = ["local", "local_fedora", "local_redhat", "local_centos", "local_debian", "local_ubuntu", "local_sles", "local_opensuse", "local_microsoft", "local_efi", "local_chain_hd0", "discovery"]
```

Comment:
I still don't like the hardcoding here. How about adding a simple macro to the templates rendering to expose the paths value (https://github.com/theforeman/foreman/blob/756207ea22e3b65d628114bed2d4975c3b318301/app/views/unattended/provisioning_templates/snippet/_pxegrub_chainload.erb#L8), and then use it from here, as well as from the `pxegrub_global_default.erb` and `pxegrub_chainload`

Reply:
I agree that will be less prone to errors in future, [this](https://github.com/theforeman/foreman/blob/develop/lib/foreman/renderer/scope/macros/host_template.rb) would most likely be the best file to define the constant and accessing macro, also it needs to be allowed then at [here](https://github.com/theforeman/foreman/blob/develop/lib/foreman/renderer/configuration.rb#L4)

Reply:
I wonder if that's an overkill, I mean macro in core just for this. Also regression in this is not fatal. If you insist...

Reply:
Alternative approaches: A) Use a shared variable in template and snippet or B) Reorder items so the chainloading EFI goes last (and ignore it - not available through the setting).

Reply:
I'd be fine with A) if we don't use the list elsewhere

Reply:
The problem with (A) is that it is a breaking change, users will need to update their templates. Unless we provide some default value which gives us back to the start.

Reply:
Since it's just PXEGrub which is not widely used, I am going for it. Works fine.